### PR TITLE
Added resources limits definition for wait-for-db container

### PR DESCRIPTION
**Pull request by Shelestov7**
# What this PR does
Added 'resources limits' definition for wait-for-db container

## Which issue(s) this PR fixes

I face a problem: when i install OnCall by Helm, my pods with oncall-engine and oncall-celery stuck on Init state, because they don't have enough resources to run.

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add debounce on Select UI components to avoid making API search requests on each key-down event by
   @maskin25 ([#2466](https://github.com/grafana/oncall/pull/2466))
 
+- Added 'resources limits' definition for wait-for-db in init container 
+
```
Comment (joeyorlando) [inline CHANGELOG.md:20]: Do you mind moving this above `## v1.3.9 (2023-07-12)` as such:
```markdown
## Unreleased

### Added

- [Helm] Added ability to specify `resources` definition within the `wait-for-db` init container by @Shelestov7
  ([#2501](https://github.com/grafana/oncall/pull/2501))

```

File: helm/oncall/templates/_helpers.tpl [hunk 2]
```diff
@@ -92,6 +92,8 @@ Create the name of the service account to use
   command: ['sh', '-c', "until (python manage.py migrate --check); do echo Waiting for database migrations; sleep 2; done"]
   securityContext:
   {{ toYaml .Values.init.securityContext | nindent 4 }}
+  resources:
+  {{ toYaml .Values.init.securityContext | nindent 4 }}
```
Comment (joeyorlando) [inline helm/oncall/templates/_helpers.tpl:96]:
```suggestion
  {{ toYaml .Values.init.resources | nindent 4 }}
```

File: helm/oncall/templates/_helpers.tpl [hunk 3]
```diff
@@ -107,6 +109,8 @@ Create the name of the service account to use
...
```
Comment (joeyorlando) [inline helm/oncall/templates/_helpers.tpl:113]:
```suggestion
  {{ toYaml .Values.init.resources | nindent 4 }}
```

File: helm/oncall/tests/__snapshot__/wait_for_db_test.yaml.snap
Comment (joeyorlando) [inline helm/oncall/tests/__snapshot__/wait_for_db_test.yaml.snap:1]: do you mind adding some unit tests to `helm/oncall/tests/wait_for_db_test.yaml`? Ideally they'd assert that using `init.resources` results in the correct output.

**Review by joeyorlando — COMMENTED** [review]
thanks for your contribution 🙂 left some minor comments. Once those are addressed, I can ✅

File: helm/oncall/tests/__snapshot__/wait_for_db_test.yaml.snap
Comment (Shelestov7) [reply helm/oncall/tests/__snapshot__/wait_for_db_test.yaml.snap:1]: You mean add test data to the snapshot?

Comment (Shelestov7) [reply helm/oncall/tests/__snapshot__/wait_for_db_test.yaml.snap:1]: Got it)

Comment (joeyorlando) [reply helm/oncall/tests/__snapshot__/wait_for_db_test.yaml.snap:1]: see the current tests in `helm/oncall/tests/wait_for_db_test.yaml`. I would add two new ones that verify for both mysql and postgresql, the resources object gets properly set

File: helm/oncall/tests/wait_for_db_test.yaml [hunk 4]
```diff
@@ -8,6 +8,14 @@ chart:
...
 tests:
   - it: database.type=mysql -> should create initContainer for MySQL database (default)
+    set:
+      resources:
+            limits:
+              cpu: 100m
+              memory: 128Mi
+            requests:
+              cpu: 100m
+              memory: 128Mi
```
Comment (joeyorlando) [inline helm/oncall/tests/wait_for_db_test.yaml:18]:
```suggestion
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
```

File: helm/oncall/tests/wait_for_db_test.yaml [hunk 5]
```diff
@@ -26,6 +34,13 @@ tests:
     set:
       database.type: postgresql
       externalPostgresql.host: some-postgresql-host
+      resources:
+            limits:
+              cpu: 100m
+              memory: 128Mi
+            requests:
+              cpu: 100m
+              memory: 128Mi
```
Comment (joeyorlando) [inline helm/oncall/tests/wait_for_db_test.yaml:43]:
```suggestion
      resources:
        limits:
          cpu: 100m
          memory: 128Mi
        requests:
          cpu: 100m
          memory: 128Mi
```

**Review by joeyorlando — COMMENTED** [review]
thanks for adding tests! one last minor suggestion w/ the indentation

File: helm/oncall/tests/wait_for_db_test.yaml (hunk 5 above)
Comment (Shelestov7) [reply helm/oncall/tests/wait_for_db_test.yaml]: FIXED

File: helm/oncall/tests/wait_for_db_test.yaml [hunk 6]
```diff
@@ -8,6 +8,14 @@ chart:
   appVersion: v1.2.36
 tests:
   - it: database.type=mysql -> should create initContainer for MySQL database (default)
+    set:
+      resources:
```
Comment (joeyorlando) [inline helm/oncall/tests/wait_for_db_test.yaml:12]: same here
```suggestion
      init.resources:
```

**Review by joeyorlando — COMMENTED** [review]
you'll need to rerun the tests locally and update the `wait_for_db_test.yaml.snap` file to reflect the new changes. That should be the last thing 🙂

File: helm/oncall/tests/wait_for_db_test.yaml (hunk 6 above)
Comment (Shelestov7) [reply helm/oncall/tests/wait_for_db_test.yaml]: You were right. Fixed it.

File: helm/oncall/tests/wait_for_db_test.yaml [hunk 7]
```diff
@@ -26,6 +34,13 @@ tests:
     set:
       database.type: postgresql
       externalPostgresql.host: some-postgresql-host
+      resources:
```
[the remark this answers is not in the document]
Comment (Shelestov7) [reply helm/oncall/tests/wait_for_db_test.yaml]: You were right. Fixed it.

**Review by joeyorlando — APPROVED** [review]

**Comment by Shelestov7** [comment]
2 checks was Failed i have no idea why.

**Pull request merged**